pysat/_instrument.py

```diff
@@ -458,16 +464,21 @@ def __getitem_xarray__(self, key):
             inst[datetime1:datetime1, 'name1':'name2']
 
         """
-        if 'time' not in self.data:
+        if 'Epoch' in self.data.indexes:
+            epoch_name = 'Epoch'
+        elif 'time' in self.data.indexes:
+            epoch_name = 'time'
+        else:
             return xr.Dataset(None)
+
         if isinstance(key, tuple):
             if len(key) == 2:
                 # support slicing time, variable name
                 try:
-                    return self.data.isel(time=key[0])[key[1]]
+                    return self.data.isel(indexers={epoch_name: key[0]})[key[1]]
                 except:
```

Comment:
`except` needs an exception type, or change this to an `if/else` statement.

Reply:
I'm all for improving this area for v3.0. I will note that we may want to be careful in changing the data setters and getters as a final code change in v2.2. While we have testing we may unintentionally change a behavior and not quite know.

Reply:
I have yet another branch as part of flake8 that addresses the bare except statements.  These should be `except (TypeError, KeyError)`.  Was planning on addressing that in 3.0

Reply:
For some context about my comment above while there is also a change in the method - I'll note that in a previous commit I changed the index name for xarray stuff to 'Epoch' from 'time'. I did this to keep consistency with pandas and avoid other potential issues (issues I just recently became aware of). One downside of that change is it broke downstream packages. The accommodation was to then support both 'Epoch' and 'time' until we reached a good point for a breaking change with a good plan.

Reply:
Thanks @jklenzing I didn't know you had already sorted things out.

Reply:
It's just buried in a branch waiting on us to merge these in.  Granted, it's part of the 3.0 milestones.  I could fix the bare except statements here if needed.

Reply:
If this or any other of my comments are addressed elsewhere, I don't care if they're addressed here.